Foundation/FileManager.swift

```diff
@@ -120,18 +120,320 @@ open class FileManager : NSObject {
         return result
     }
     
+    private enum _SearchPathDomain {
+        case system
+        case local
+        case network
+        case user
+        
+        init?(_ domainMask: SearchPathDomainMask) {
+            if domainMask == .systemDomainMask {
+                self = .system
+            }
+            if domainMask == .localDomainMask {
+                self = .local
+            }
+            if domainMask == .networkDomainMask {
+                self = .network
+            }
+            if domainMask == .userDomainMask {
+                self = .user
+            }
+            
+            return nil
```

Comment:
Wont this failable initialiser always fail?

Reply:
How so? It didn’t in local testing. (It will fail for bitmasks with more than one bit set and that’s intended.)

Reply:
Oh dear. Yeah, I see the problem.

Reply:
This patch needs some care and attention.